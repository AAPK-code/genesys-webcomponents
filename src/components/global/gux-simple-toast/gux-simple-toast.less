gux-simple-toast {
  > div {
    padding: 12px 12px;
    display: flex;
    width: 336px;
    position: relative;
    border: 1px solid;
    .content-wrapper {
      flex: 1;
      overflow: hidden;
      line-height: 16px;
      padding-right: 40px;
    }
    .icon-wrapper {
      height: 16px;
    }
    .icon-wrapper i,
    .icon-wrapper img {
      font-size: 16px;
      height: 16px;
      width: 16px;
      margin-right: 8px;
    }
    .message {
      overflow: hidden;
      text-overflow: ellipsis;
      max-height: 32px;
    }
    .close-icon-wrapper {
      position: absolute;
      top: 12px;
      right: 12px;
      button {
        cursor: pointer;
        border: none;
        background: none;
      }
    }
  }
}

// Theming

// Dark
.gux-simple-toast-dark-theme {
  > div {
    background: @gux-off-white;
    color: @gux-grey-2;
    border-color: @gux-border;
    .close-icon-wrapper {
      button {
        color: @gux-pale-gray;
      }
    }
    .icon-wrapper.alert {
      color: @gux-alert;
    }
    .icon-wrapper.warning {
      color: @gux-warning;
    }
    .icon-wrapper.positive {
      color: @gux-positive;
    }
    .icon-wrapper.neutral {
      color: @gux-neutral;
    }
  }
}

.gux-dark-theme {
  gux-simple-toast {
    .gux-simple-toast-dark-theme();
  }
}
gux-simple-toast.gux-dark-theme {
  .gux-simple-toast-dark-theme();
}

// Light
.gux-simple-toast-light-theme {
  > div {
    background: @gux-panel;
    color: @gux-off-white;
    border-color: @gux-grey-2;
    .close-icon-wrapper {
      button {
        color: @gux-pale-gray-2;
      }
    }
    .icon-wrapper.alert {
      color: @gux-alert-2;
    }
    .icon-wrapper.warning {
      color: @gux-warning;
    }
    .icon-wrapper.positive {
      color: @gux-positive;
    }
    .icon-wrapper.neutral {
      color: @gux-neutral-2;
    }
  }
}

.gux-light-theme {
  gux-simple-toast {
    .gux-simple-toast-light-theme();
  }
}
gux-simple-toast.gux-light-theme {
  .gux-simple-toast-light-theme();
}
// Default Theme
gux-simple-toast {
  .gux-simple-toast-light-theme();
}
