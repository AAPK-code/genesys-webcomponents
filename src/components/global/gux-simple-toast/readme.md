# gux-simple-toast

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. 

<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description                                               | Type     | Default     |
| ------------ | ------------- | --------------------------------------------------------- | -------- | ----------- |
| `accent`     | `accent`      | The component accent (alert warning positive or neutral). | `string` | `'neutral'` |
| `closeLabel` | `close-label` | The label for the close button                            | `string` | `''`        |
| `icon`       | `icon`        | The icon name of the title.                               | `string` | `undefined` |
| `iconUri`    | `icon-uri`    | The icon uri of the title.                                | `string` | `undefined` |
| `idToast`    | `id-toast`    | The id of the title.                                      | `string` | `undefined` |
| `message`    | `message`     | The message of the toast.                                 | `string` | `undefined` |


## Events

| Event        | Description | Type               |
| ------------ | ----------- | ------------------ |
| `closeClick` |             | `CustomEvent<any>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
