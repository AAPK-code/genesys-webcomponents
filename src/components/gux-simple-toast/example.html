<gux-simple-toast message="Simple message"></gux-simple-toast>
