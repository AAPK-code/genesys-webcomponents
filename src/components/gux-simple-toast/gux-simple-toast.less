gux-simple-toast {
  padding: 8px 12px;
  width: 336px;
  position: relative;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-content: stretch;
  align-items: flex-start;
  border-top: 1px solid;

  .icon{
    order: 0;
    flex: 0 1 auto;
    align-self: auto;
    margin: 6px;

    &.alert {
      color: @gux-alert;
    }
    &.warning {
      color: @gux-warning;
    }
    &.positive {
      color: @gux-positive;
    }
    &.neutral {
      color: @gux-neutral;
    }

    gux-icon {
      width: 12px
    }
  }

  .message{
    order: 0;
    flex: 1 1 auto;
    align-self: auto;
    line-height: 24px;
    margin: 0 12px 0 8px;
    overflow: hidden;
    text-overflow: ellipsis;
    max-height: 48px;
  }

  .dismiss-button {
    position: absolute;
    top: 15px;
    right: 15px;
    padding: 0;
    border: none;
    background: transparent;
    font-size: 11px;
    cursor: pointer;
    gux-icon {
      width: 12px;
    }
  }
}

// Theming
// Light
.gux-simple-toast-light-theme {
  background: @gux-panel;
  color: @gux-off-white;
  border-color: @gux-grey-2;

  .icon{
    &.alert {
      color: @gux-alert-2;
    }
    &.warning {
      color: @gux-warning;
    }
    &.positive {
      color: @gux-positive;
    }
    &.neutral {
      color: @gux-neutral-2;
    }
  }

  .dismiss-button {
    color: @gux-off-white;
  }
}

// Dark
.gux-simple-toast-dark-theme {
  background: @gux-off-white;
  color: @gux-grey-2;
  border-color: @gux-border;

  .icon {
    &.alert {
      color: @gux-alert;
    }
    &.warning {
      color: @gux-warning;
    }
    &.positive {
      color: @gux-positive;
    }
    &.neutral {
      color: @gux-neutral;
    }
  }

  .dismiss-button {
    color: @gux-grey-2;
  }
}

// Default Theme
gux-simple-toast {
  .gux-simple-toast-light-theme();
}

// Light Theme
.gux-light-theme {
  gux-simple-toast {
    .gux-simple-toast-light-theme();
  }
}
gux-simple-toast.gux-light-theme {
  .gux-simple-toast-light-theme();
}

// Dark Theme
.gux-dark-theme {
  gux-simple-toast {
    .gux-simple-toast-dark-theme();
  }
}
gux-simple-toast.gux-dark-theme {
  .gux-simple-toast-dark-theme();
}
