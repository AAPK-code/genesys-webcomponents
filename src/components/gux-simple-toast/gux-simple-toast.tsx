import { Component, Event, EventEmitter, h, Prop } from '@stencil/core';

@Component({
  styleUrl: 'gux-simple-toast.less',
  tag: 'gux-simple-toast'
})
export class GuxSimpleToast {
  @Event()
  closeClick: EventEmitter;

  /**
   * The component accent (alert warning positive or neutral).
   */
  @Prop()
  accent: string = 'neutral';

  /**
   * The id of the title.
   */
  @Prop()
  idToast: string;
  /**
   * The icon name of the title.
   */
  @Prop()
  icon: string;
  /**
   * The icon uri of the title.
   */
  @Prop()
  iconUri: string;
  /**
   * The message of the toast.
   */
  @Prop()
  message: string;

  /**
   * The label for the close button
   */
  @Prop()
  closeLabel: string = '';

  onCloseButtonClickHandler(e) {
    this.closeClick.emit({ idToast: this.idToast, originalEvent: e });
  }

  /**
   * This function is to check accent
   */
  getAccent() {
    const accent = this.accent.toLowerCase();
    const allowed = ['warning', 'alert', 'neutral', 'positive'];
    return allowed.includes(accent) ? accent : 'neutral';
  }

  getIcon() {
    if (this.icon) {
      return <i class={this.icon} />;
    } else if (this.iconUri) {
      return <img src={this.iconUri} />;
    }
    return '';
  }

  render() {
    return (
      <div class="gux-simple-toast">
        <div class={'icon-wrapper ' + this.getAccent()}>{this.getIcon()}</div>
        <div class="content-wrapper">
          <div class="message" title={this.message}>
            {this.message}
          </div>
        </div>
        <div class="close-icon-wrapper">
          <button
            onClick={e => this.onCloseButtonClickHandler(e)}
            title={this.closeLabel}
          >
            <gux-icon decorative iconName="ic-close"></gux-icon>
          </button>
        </div>
      </div>
    );
  }
}
