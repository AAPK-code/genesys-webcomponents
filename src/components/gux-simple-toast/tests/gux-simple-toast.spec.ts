import { GuxSimpleToast } from '../gux-simple-toast';

describe('gux-simple-toast', () => {
  it('builds', () => {
    expect(new GuxSimpleToast()).toBeTruthy();
  });

  describe('getAccent', () => {
    it('should return neutral if props/attr is not set correctly', () => {
      const component = new GuxSimpleToast();
      expect(component.getAccent()).toBe('neutral');
      component.accent = 'aaaaaa';
      expect(component.getAccent()).toBe('neutral');
    });
    it('should return accent in lowercase if props/attr is set correctly', () => {
      const component = new GuxSimpleToast();
      expect(component.getAccent()).toBe('neutral');
      component.accent = 'PoSitive';
      expect(component.getAccent()).toBe('positive');
    });
  });

  describe('getIcon', () => {
    it('should return icon and accent classname', () => {
      const component = new GuxSimpleToast();
      expect(component.getIcon()).toBe('');
      component.iconUri = 'test';
      // testing JSX
      expect((component.getIcon() as any).$tag$).toBe('img');
      expect((component.getIcon() as any).$attrs$.src).toBe('test');
      component.icon = 'test';
      // testing JSX
      expect((component.getIcon() as any).$tag$).toBe('gux-icon');
      expect((component.getIcon() as any).$attrs$.iconName).toBe('test');
    });
  });
});
