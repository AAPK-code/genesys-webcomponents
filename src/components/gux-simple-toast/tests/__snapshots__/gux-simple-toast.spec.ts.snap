// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-simple-toast #render should render alert simple toast 1`] = `
<gux-simple-toast accent="alert">
  <!---->
  <div class="alert icon">
    <gux-icon decorative="" icon-name="add-user" slot="icon"></gux-icon>
  </div>
  <div class="message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <button class="dismiss-button" title="Dismiss">
    <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
  </button>
</gux-simple-toast>
`;

exports[`gux-simple-toast #render should render neutral simple toast 1`] = `
<gux-simple-toast accent="neutral">
  <!---->
  <div class="icon neutral">
    <gux-icon decorative="" icon-name="add-user" slot="icon"></gux-icon>
  </div>
  <div class="message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <button class="dismiss-button" title="Dismiss">
    <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
  </button>
</gux-simple-toast>
`;

exports[`gux-simple-toast #render should render positive simple toast 1`] = `
<gux-simple-toast accent="positive">
  <!---->
  <div class="icon positive">
    <gux-icon decorative="" icon-name="add-user" slot="icon"></gux-icon>
  </div>
  <div class="message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <button class="dismiss-button" title="Dismiss">
    <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
  </button>
</gux-simple-toast>
`;

exports[`gux-simple-toast #render should render warning simple toast 1`] = `
<gux-simple-toast accent="warning">
  <!---->
  <div class="icon warning">
    <gux-icon decorative="" icon-name="add-user" slot="icon"></gux-icon>
  </div>
  <div class="message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <button class="dismiss-button" title="Dismiss">
    <gux-icon icon-name="ic-close" screenreader-text="Dismiss"></gux-icon>
  </button>
</gux-simple-toast>
`;
